confluent_kafka/src/Consumer.c

```diff
@@ -359,6 +381,54 @@ static PyObject *Consumer_position (Handle *self, PyObject *args,
 }
 
 
+static PyObject *Consumer_get_watermark_offsets (Handle *self, PyObject *args,
+						 PyObject *kwargs) {
+
+	TopicPartition *tp;
+	rd_kafka_resp_err_t err;
+	double tmout = -1.0f;
+	int cached = 0;
+	int64_t low = RD_KAFKA_OFFSET_INVALID, high = RD_KAFKA_OFFSET_INVALID;
+	static char *kws[] = { "partition", "timeout", "cached", NULL };
+	PyObject *rlist;
+
+	if (!PyArg_ParseTupleAndKeywords(args, kwargs, "O|db", kws,
+					 (PyObject **)&tp, &tmout, &cached))
+		return NULL;
+
+
+	if (PyObject_Type((PyObject *)tp) != (PyObject *)&TopicPartitionType) {
+		PyErr_Format(PyExc_TypeError,
+			     "expected %s", TopicPartitionType.tp_name);
+		return NULL;
+	}
+
+	if (cached) {
+		err = rd_kafka_get_watermark_offsets(self->rk,
+						     tp->topic, tp->partition,
+						     &low, &high);
+	} else {
+		err = rd_kafka_query_watermark_offsets(self->rk,
+						       tp->topic, tp->partition,
+						       &low, &high,
+						       tmout >= 0 ? (int)(tmout * 1000.0f) : -1);
+	}
+
+	if (err) {
+		cfl_PyErr_Format(err,
+				 "Failed to get watermark offsets: %s",
+				 rd_kafka_err2str(err));
+		return NULL;
+	}
+
+	rlist = PyList_New(2);
```

Comment:
Since it is fixed length anyway, this would normally be a tuple. (And docs below would need update too.)